pandas/core/indexes/base.py

```diff
@@ -800,24 +800,29 @@ def __deepcopy__(self, memo=None):
             memo = {}
         return self.copy(deep=True)
 
-    def _validate_names(self, name=None, names=None, deep=False):
+    def _handle_names(self, name=None, names=None, deep=False):
         """
         Handles the quirks of having a singular 'name' parameter for general
         Index and plural 'names' parameter for MultiIndex.
         """
         from copy import deepcopy
-        if names is not None and name is not None:
-            raise TypeError("Can only provide one of `names` and `name`")
-        elif names is None and name is None:
+
```

Comment:
is there a reason you are splitting this into 2 functions? what is wrong with the original that this is needed?

Reply:
Mostly because we need validation logic not only for copying (see https://github.com/pandas-dev/pandas/pull/19168). It should reduce repetitions in the code.

Reply:
ok, not averse to this, maybe its the spelling and how these are split. if you can clearly define each purpose with a doc-string then all for it.